homeassistant/components/hue/light.py

```diff
@@ -378,7 +380,7 @@ def device_info(self):
         elif effect == EFFECT_RANDOM:
             command['hue'] = random.randrange(0, 65535)
             command['sat'] = random.randrange(150, 254)
-        elif self.is_philips:
+        elif self.is_philips or effect == '':
```

Comment:
This is weird. Can you test sending `effect = None` when there is no active effect on osram bulbls? If it works in the latest firmware, we can drop this check. We should also only set the effect command when `ATTR_EFFECT in kwargs`. Now we will always set it to none, even if the user didn't pass in anything (probably 99% of the time).

Reply:
Current implementation does not send `effect ='none'` to non philips bulbs. That's good for older osram bulbs, but necessary to be send if you have an osram bulb with current firmware. Otherwise you cannot remove the effect. Only removing from the socket works.

This `elif` is to allow manually sending a 'none' to a non philips bulb if ATTR_EFFECT in kwargs is `''`. Normally effect is `None`, so this `elif` doesn't matter.

Switching from `elif self.is_philips` to `else` may breake implementation for bulbs without latest firmware.

Reply:
But that is assuming that an empty string is the way of disabling the effect. As far as I know, we don't have anything defined, but generally we use `None`. Because we use `kwargs.get(`, we cannot differentiate between no effect passed in vs an empty effect passed in. We can do that by adding an `ATTR_EFFECT in kwargs` check.